app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/TravelCertificateHistory.kt

```diff
@@ -0,0 +1,422 @@
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(16.dp, 16.dp, 16.dp, 8.dp),
+    )
+    HedvigSecondaryContainedButton(
+      text = stringResource(R.string.travel_certificate_get_travel_certificate_button),
+      onClick = onContinue,
+      modifier = Modifier.padding(16.dp, 8.dp, 16.dp, 0.dp),
+    )
+    Spacer(Modifier.height(32.dp))
+  }
+}
+
+@Composable
+private fun ShowInitialInfo() {
+  Column(
+    modifier = Modifier.fillMaxSize(),
+    horizontalAlignment = Alignment.CenterHorizontally,
+    verticalArrangement = Arrangement.Center,
+  ) {
+    HedvigInformationSection(
+      stringResource(id = R.string.travel_certificate_empty_list_message),
+    )
+  }
+}
+
+@Composable
+private fun ShowNotEmptyList(list: List<TravelCertificate>, onCertificateClick: (String) -> Unit) {
+  Column(
+    horizontalAlignment = Alignment.CenterHorizontally,
+  ) {
+    for (i in list) {
+      CertificateCard(
+        certificate = i,
+        onCertificateClick = onCertificateClick,
+      )
+    }
+  }
+}
+
+@Composable
+private fun CertificateCard(
+  certificate: TravelCertificate,
+  onCertificateClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  Row(
+    verticalAlignment = Alignment.CenterVertically,
+    modifier = modifier
+      .fillMaxWidth()
+      .clickable(onClick = { onCertificateClick(certificate.signedUrl) })
+      .padding(16.dp),
+  ) {
+    val isExpired = isExpired(certificate.expiryDate)
+    val color = if (isExpired) Color.Red else Color.Unspecified
+    val expiryDateText =
+      "${certificate.expiryDate.dayOfMonth} ${certificate.expiryDate.month.name.lowercase().substring(0, 3)}"
+    Text(
+      text = expiryDateText,
+      style = MaterialTheme.typography.bodyLarge,
+      color = color,
+    )
+    Spacer(modifier = Modifier.weight(1f))
+    Text(
+      text = if (isExpired) {
+        stringResource(id = R.string.travel_certificate_expired)
+      } else {
+        stringResource(id = R.string.travel_certificate_active)
+      },
+      style = MaterialTheme.typography.bodyLarge,
+      color = color,
+    )
+  }
+}
+
+private fun isExpired(expiryDate: LocalDate): Boolean {
+  val current = java.time.LocalDate.now()
+  val currentYear = current.year
+  val currentDay = current.dayOfYear
+  val expiryYear = expiryDate.year
+  val expiryDay = expiryDate.dayOfYear
+  return if (expiryYear >= currentYear) {
+    expiryDay <= currentDay
+  } else {
+    true
+  }
+}
+
+@HedvigPreview
+@Composable
+private fun PreviewTravelCertificateHistoryScreenWithEmptyList() {
+  HedvigTheme {
+    Surface(color = MaterialTheme.colorScheme.background) {
+      TravelCertificateHistoryScreen(
+        {},
+        {},
+        {},
+        {},
+        {},
+        CertificateHistoryUiState.SuccessDownloadingHistory(listOf()),
+      )
+    }
+  }
+}
+
+@HedvigPreview
+@Composable
+private fun PreviewTravelCertificateHistoryScreenWithExpiredEarlier() {
```

Comment:
To be able to in one preview get 2 versions out of it, we got this https://github.com/HedvigInsurance/android/blob/fbf254341f75ef46dc66514ee1ebe8627a94b0c9/app/core/core-ui/src/main/kotlin/com/hedvig/android/core/ui/SelectIndicationCircle.kt#L39-L41 which allows you to get a boolean, once false and once true.
Then you can use that to in one preview composable define what the UI should look like with the boolean in both cases. Here it could be 
`@PreviewParameter(BooleanCollectionPreviewParameterProvider::class) hasCertificateHistory: Boolean,` and then check for `hasCertificateHistory` to show the history or not.

Reply:
Aha and I see now that this has more previews than I first thought. 
In that case, you can go full out and create your own class that you provide all the possible cases for, and then you can keep one preview composable still I think.

Look here https://github.com/HedvigInsurance/android/blob/3b9a24743d46e0c5d527e8c5b16060ccf4121b38/app/feature/feature-chat/src/main/kotlin/com/hedvig/android/feature/chat/ui/ChatDestination.kt#L185-L241 for inspiration. LMK if this does not make much sense